language/move-binary-format/src/file_format.rs

```diff
@@ -1666,13 +1646,8 @@ impl CompiledScriptMut {
 /// It is a unit of code that can be used by transactions or other modules.
 ///
 /// A module is published as a single entry and it is retrieved as a single blob.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct CompiledModule(CompiledModuleMut);
-
-/// A mutable version of `CompiledModule`. Converting to a `CompiledModule` requires this to pass
-/// the bounds checker.
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
-pub struct CompiledModuleMut {
+pub struct CompiledModule {
```

Comment:
I know it was sort of the whole point of this migration, but when I saw this PR I was a bit skittish of seeing everything being mutable. It's one thing to say we want it done and another to actually see it lol

But after rethinking about it, I think most uses of a CompiledModule are through `&CompiledModule` so the mutation isn't an issue, and it will be nice to maybe slowly kill off the `ModuleAccess` trait.

Any additional thoughts @vgao1996? Feel free to approve it once you take a look

Reply:
Ah, looks like @vgao1996 is on PTO until next week. Up to you @tnowacki whether to wait until Monday for this to land or to accept and have his review ex post facto.

Reply:
Let's just land it. I think this will all feel better as we clean up more and more. I'd love to see the `ModuleAccess` trait die and everything in `language/move-binary-format/src/views.rs` die or get moved into the prover 

This was a cleanup long planned by Dario and myself, so should be fun to pick away at it (I might even pitch in just for the fun so we should sync up)